### PR TITLE
add configuration parameter "classifications" for respective cucumber-reporting config

### PR DESCRIPTION
**Pull request by KLBonn**
Hi Damian,
as suggested here my PR for classifications.
I tried to keep it as simple and "minor" as possible.
(e.g. no additional parameter "classificationsFile" to read key-value attributes from)

What do you think?
Best regards,
Klaus

PS: Key will be capitalised, because it will look nicer if you supply
<configuration>
    <classifications>
        <lower.case.key>Some arbitrary value</lower.case.key>
    </classifications>
    ...
</configuration>

### CODE REVIEW

File: src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java [hunk 1]
```diff
@@ -86,6 +95,11 @@ public void execute() throws MojoExecutionException {
             Configuration configuration = new Configuration(outputDirectory, projectName);
             configuration.setBuildNumber(buildNumber);
             configuration.setParallelTesting(parallelTesting);
+            if (classifications != null) {
```
Comment (damianszczepanik) [inline src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java]: Validate it against not null but also if is empty

File: src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java [hunk 2]
```diff
@@ -51,21 +53,28 @@
     private File cucumberOutput;
 
     /**
-     * Skip check for failed build result
+     * Skip check for failed build result.
```
Comment (damianszczepanik) [inline src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java]: Nice

File: src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java (hunk 1 above)
Comment (KLBonn) [inline src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java]: I thought when it is empty then the "for" loop will just be not executed (.entrySet() being empty, i.e. loop will be skipped immediately)?

Comment (damianszczepanik) [inline src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java]: It does. On the other hand it's good to validate both at once

**Comment by KLBonn** [comment]
Better now?

File: src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java [hunk 3]
```diff
@@ -86,15 +95,18 @@ public void execute() throws MojoExecutionException {
             Configuration configuration = new Configuration(outputDirectory, projectName);
             configuration.setBuildNumber(buildNumber);
             configuration.setParallelTesting(parallelTesting);
+            if (classifications != null && !classifications.isEmpty()) {
```
Comment (damianszczepanik) [inline src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java]: why not at once https://commons.apache.org/proper/commons-collections/javadocs/api-3.2.2/org/apache/commons/collections/CollectionUtils.html#isEmpty(java.util.Collection) ?

File: src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java [hunk 4]
```diff
@@ -86,15 +95,18 @@ public void execute() throws MojoExecutionException {
...
             ReportBuilder reportBuilder = new ReportBuilder(list, configuration);
             getLog().info("About to generate Cucumber report.");
             Reportable report = reportBuilder.generateReports();
 
-            if (checkBuildResult) {
-                if (report == null) {
-                    throw new MojoExecutionException("BUILD FAILED - Check Report For Details");
-                }
+            if (checkBuildResult && report == null) {
+                throw new MojoExecutionException("BUILD FAILED - Check Report For Details");
```
Comment (damianszczepanik) [inline src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java]: also squash changes into single commit before updating this PR

Comment (KLBonn) [inline src/main/java/net/masterthought/cucumber/CucumberReportGeneratorMojo.java]: done...

**Comment by damianszczepanik** [comment]
Thanks!

**Pull request merged**